services/common/src/components/common/ScrollSideMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC, useEffect } from "react";
+import { useParams, useHistory, useLocation } from "react-router-dom";
+import { Anchor } from "antd";
+import { ISideMenuOption } from "../../interfaces/common/sideMenuOption.interface";
+
+/**
+ * @constant ScrollSideMenu renders react children with an active indicator if the id is in the url.
+ */
+
+interface ScrollSideMenuProps {
+  menuOptions: ISideMenuOption[];
+  featureUrlRoute: any;
+  featureUrlRouteArguments: (string | number)[];
+  tabSection?: string;
+}
+
+export const ScrollSideMenu: FC<ScrollSideMenuProps> = ({ tabSection = "", ...props }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const { tab } = useParams<{
+    tab: string;
+  }>();
+  let urlRoute = undefined;
+
+  const updateUrlRoute = (route) => {
+    urlRoute = props.featureUrlRoute(...props.featureUrlRouteArguments, route);
+    if (route === history.location.hash) {
+      return;
+    }
+
+    history.push(urlRoute, { currentActiveLink: route });
+  };
+
+  useEffect(() => {
+    // If the user loads the page with a hash in the URL, start them off at the corresponding feature section.
```

Comment:
Oh man 😓 .  Nice figuring!

Reply:
Yeah this part is a bit confusing, can't take the credit though! A previous developer had worked on this and figured it out :)